src/PHP.php

```diff
@@ -0,0 +1,519 @@
+
+			if ( ! $launch->return_code ) {
+				$this->site_data['db_host'] = trim( $launch->stdout, "\n" );
+			} else {
+				throw new \Exception( 'There was a problem inspecting network. Please check the logs' );
+			}
+		}
+		\EE::log( 'Verifying connection to remote database' );
+
+		if ( ! \EE::exec( sprintf( "docker run -it --rm --network='%s' mysql sh -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='EXIT'\"", $this->site_data['site_url'], $this->site_data['db_host'], $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'] ) ) ) {
+			throw new \Exception( 'Unable to connect to remote db' );
+		}
+
+		\EE::success( 'Connection to remote db verified' );
+	}
+
+	/**
+	 * Function to create the site.
+	 */
+	private function create_site( $assoc_args ) {
+
+		$this->site_data['site_fs_path'] = WEBROOT . $this->site_data['site_url'];
+		$this->level                     = 1;
+		try {
+			\EE\Site\Utils\create_site_root( $this->site_data['site_fs_path'], $this->site_data['site_url'] );
+			$this->level = 2;
+
+			$containers = [ 'nginx', 'postfix' ];
+
+			if ( ! empty( $assoc_args['with-db'] ) && 'db' === $this->site_data['db_host'] ) {
+				$this->maybe_verify_remote_db_connection();
+				$containers[] = 'db';
+			}
+			$this->level = 3;
+			$this->configure_site_files();
+
+			\EE\Site\Utils\start_site_containers( $this->site_data['site_fs_path'], $containers );
+			\EE\Site\Utils\configure_postfix( $this->site_data['site_url'], $this->site_data['site_fs_path'] );
+
+			\EE\Site\Utils\create_etc_hosts_entry( $this->site_data['site_url'] );
+			if ( ! $this->skip_status_check ) {
+				$this->level = 4;
+				\EE\Site\Utils\site_status_check( $this->site_data['site_url'] );
+			}
+
+			\EE\Site\Utils\add_site_redirects( $this->site_data['site_url'], false, 'inherit' === $this->site_data['site_ssl'] );
+			\EE\Site\Utils\reload_proxy_configuration();
+
+			if ( $this->site_data['site_ssl'] ) {
+				$wildcard = $this->site_data['site_ssl_wildcard'];
+				\EE::debug( 'Wildcard in site php command: ' . $this->site_data['site_ssl_wildcard'] );
+				$this->init_ssl( $this->site_data['site_url'], $this->site_data['site_fs_path'], $this->site_data['site_ssl'], $wildcard );
+
+				\EE\Site\Utils\add_site_redirects( $this->site_data['site_url'], true, 'inherit' === $this->site_data['site_ssl'] );
+				\EE\Site\Utils\reload_proxy_configuration();
+			}
+		} catch ( \Exception $e ) {
+			$this->catch_clean( $e );
+		}
+
+		$this->info( [ $this->site_data['site_url'] ], [] );
+		$this->create_site_db_entry();
+	}
+
+	/**
+	 * Function to save the site configuration entry into database.
+	 */
+	private function create_site_db_entry() {
+		$ssl = null;
+
+		if ( $this->site_data['site_ssl'] ) {
+			$ssl = 'letsencrypt';
+		}
+
+		$data = [
+			'site_url'             => $this->site_data['site_url'],
+			'site_type'            => $this->site_data['site_type'],
```

Comment:
Add:

'app_sub_type' => 'mysql/with_db'  if  `--with-db` flag is passed else fill it with 'php'.

Then change all the:
 `if ( ! empty( $this->site_data['db_host'] ) )` checks with: 
```
if ( 'mysql/with_db' === $this->site_data['app_sub_type'] ) )
```